### ISSUE TITLE
Understanding   Concurrent Input Mechanisms

### ISSUE THREAD

**Issue by michael-n-cooper** [w3c/wcag#356]
_From @awkawk on March 2, 2018 15:24_

<p>The <a   href="https://www.w3.org/TR/WCAG21/#concurrent-input-mechanisms"   target="_blank">2.5.4 Concurrent Input Mechanisms (AAA)   SC</a> has draft Understanding content that can be viewed at <a   href="https://rawgit.com/w3c/wcag21/concurrent-input-mechanisms/understanding/21/concurrent-input-mechanisms.html"   target="_blank">Understanding Concurrent Input   Mechanisms</a>.</p><p>Please review the Understanding   document and submit a comment on this issue for any changes that are needed   to the content.</p>

_Copied from original issue: w3c/wcag21#792_

**Comment by michael-n-cooper** [comment]
_From @joshueoconnor on March 8, 2018 17:13_

Some suggestions:

'Users $may employ a variety of input mechanisms when interacting with web content. $This may be a combination of mechanisms such as a keyboard, keyboard-like interfaces and pointer devices like  mouse, stylus or touchscreen.

'Even though a device may have a primary or preferred input mechanism, the user $may choose to apply alternative mechanisms. $For example, a primary mechanism is the touchscreen on mobile phones and tablets while a paired mouse or external keyboard $may be the alternative.'

'Dynamically add, remove or switch a particular input mechanism can be done at any point by a user when certain tasks and interactions are easier to accomplish that way.

'Web content must therefore not limit the user's interaction to any $singular input mechanism unless the restriction is essential, or required to ensure the security of the content, or  respect user settings.'

**Comment by michael-n-cooper** [comment]
_From @mbgower on March 8, 2018 23:11_

> Users are allowed to add and remove input mechanisms at any point.

This sounds like an OS feature. I think it can be qualified like this:
> Users are allowed to add and remove input mechanisms at any point, where supported by the operating system

**Comment by michael-n-cooper** [comment]
_From @mbgower on March 8, 2018 23:40_

I made a bunch of editorial changes and submitted as [Pull Request 800](https://github.com/w3c/wcag21/pull/800). Note that I removed the following from examples, because my understanding is that Examples are normally supposed to be examples of success. I appended one to the Intent section as a note.

 <li>A touch-typing web application, which teaches users how to touch-type on a keyboard and/or measures their proficiency and speed, would be an example of an essential limitation to a particular input mechanism. In this case, the application would not be required to allow the user to switch to an alternative input mechanism.</li>
        <li>An application contains a field for the user to provide a signature. Because using a mouse, or finger on a touchscreen, usually results in something that is only vaguely reminiscent of a user's real signature, the web content restricts use of that field only to pen/stylus devices as their fidelity is high enough to allow the user to approximate their signature.</li>

**Comment by michael-n-cooper** [comment]
_From @lseeman on April 22, 2018 16:37_

i think people reading this will have no idea what to do

**Comment by michael-n-cooper** [comment]
_From @lseeman on April 22, 2018 16:38_

I would add a bit of implemention detail into the understanding so people will not worry about html controls

**Comment by michael-n-cooper** [comment]
_From @cwadamsoforacle on April 23, 2018 14:21_

I've made some changes, but I'm not a github aficionado.  I've obtained master and created a branch (locally), but my efforts to push back to github are resulting in errors.  I need some tutorial on github usage.  In any event, here are my proposed changes (which impact the first three paragraphs):

Users may employ a variety of input mechanisms when interacting with web content. These may be a combination of mechanisms such as a keyboard or keyboard-like interfaces and pointer devices like a mouse, stylus or touchscreen.

Even though a device may have a primary input mechanism, the user may choose to employ alternative input mechanisms when interacting with the device. For example, the primary mechanism for mobile phones and tablets is the touchscreen.  The user of these devices may choose to use a paired mouse or external keyboard as an alternative to the using the touchscreen.

A user must be able to switch input mechanisms at any point should the user determine that certain tasks and interactions are more easily accomplished by using an alternative input mechanism. Content must not limit the user's interaction to any particular input mechanism unless the restriction is essential, or is required to ensure the security of the content or to respect user settings.

**Comment by michael-n-cooper** [comment]
_From @goodwitch on April 23, 2018 14:23_

Hey Chuck,

Wanna jump in my Zoom Room and I'll show you what I know (to make github
your friend)?

[Alastair - removed link for privacy!]

G

**Comment by michael-n-cooper** [comment]
_From @cwadamsoforacle on April 23, 2018 15:3_

I freed up from a meeting.  I’m available now and for the next 2 hours if you are available.  I would love to get educated/smarter.

**Comment by michael-n-cooper** [comment]
_From @alastc on May 8, 2018 23:13_

Hi @cwadamsoforacle did you make any progress with this one? (Understanding Concurrent Input Mechanisms)

Thanks,

-Alastair

**Comment by michael-n-cooper** [comment]
_From @cwadamsoforacle on May 8, 2018 23:24_

I was not aware that there was any additional action required of/by me on this.  I’m not aware of what that action should be.

The fact that you are asking indicates that there is “more” that needs to be done, though I’m not sure what that “more” is.  I’m happy to do the “more” if you can educate me on what else needs to happen.

I made my suggested changes and put them in github, I think they were merged appropriately (though not by me).

Thanks,

Charles Adams

**Comment by michael-n-cooper** [comment]
_From @alastc on May 8, 2018 23:32_

Ah, great. I couldn't tell from the comments if you talked to Glenda and made some updates, I don't get an automatic ping about commits.

I'll add that to the 'ready to review' set, thanks.

**Comment by michael-n-cooper** [comment]
_From @cwadamsoforacle on May 9, 2018 14:17_

Be advised Alastair:  I THINK that’s where we are at.  This is the first time through this cycle for me, and there’s plenty I might be misinterpreting.

 

Thanks,

Charles Adams

**Comment by fstrr** [comment]
This issue looks to be inactive, so I'm closing it. Please re-open if necessary.
